### DISCUSSION TITLE
Evaluate code selection

### DISCUSSION THREAD

**Discussion by dodalovicgran** [rust-lang/rust-analyzer#15451] [Ideas]
IntelliJ IDEA rust plugin can evaluate the type of an expression

![image](https://github.com/rust-lang/rust-analyzer/assets/116893230/b372f29c-3a5c-40e4-9e9d-5e6cd245264c)

Is there anything (already or planned) in the rust-analyzer?

Thank You

**Comment by Veykril** [comment]
You can select the expression you are interested in and then hover that selection which should make r-a show the type of the selected expression

**Comment by dodalovicgran** [comment]
I think it even fails to evaulate:
![image](https://github.com/rust-lang/rust-analyzer/assets/116893230/f77f9a4e-f866-4c6a-b5ca-8431fd4ca0fe)

Although the code hints show it as `HttpServer<impl ....`

:thinking:

**Comment by Veykril** [comment]
Ah, we don't seem to render the resulting type for `?` expressions on `Result`s, only their errors 😅 We should fix that
